localstack/services/sns/sns_listener.py

```diff
@@ -659,17 +686,16 @@ def publish_message(topic_arn, req_data, headers, subscription_arn=None, skip_ch
     return message_id
 
 
-def publish_batch(topic_arn, req_data, headers):
+def publish_batch(topic_arn, messages, headers):
     response = {"Successful": [], "Failed": []}
-    messages = parse_urlencoded_data(
-        req_data, "PublishBatchRequestEntries.member", "MessageAttributes.entry"
-    )
     for message in messages:
         message_id = str(uuid.uuid4())
         data = {}
         data["TopicArn"] = [topic_arn]
         data["Message"] = [message["Message"]]
         data["Subject"] = [message["Subject"]]
+        if ".fifo" in topic_arn:
+            data["MessageGroupId"] = message.get("MessageGroupId")
 
```

Comment:
@baermat thanks for pointing this out. We need to handle the MessageDeduplicationId for FIFO topics same way as SQS FIFO queues. I left it here to be consistent with current implementation of `Publish` API  for FIFO topics (which is also missing this check). 

We can handle this missing checks for both API's either here or in separate PR. let me know your thoughts. Thanks!!!!

Reply:
Hard to say, on the one hand it might get lost on the other it should be consistent with the current code. I'd say put a comment there like "#TODO: add MessageDeduplication checks once ASF-SQS implementation becomes default" and handle it in a different PR.

Reply:
ok, sounds good. Will update the PR soon . Thanks!!!!

Reply:
Thanks @KrishnanRanjithkumar for the update. 👍  Is the PR now ready to re-review/merge? Thanks

Reply:
@whummer , I updated the PR few days ago, it's ready for re-review. Thanks